pxd.c

```diff
@@ -89,6 +89,46 @@ static void pxd_release(struct gendisk *disk, fmode_t mode)
 	put_device(&pxd_dev->dev);
 }
 
+static long pxd_ioctl_dump_fc_info(void)
+{
+	int i;
+	struct pxd_context *ctx;
+
+	for (i = 0; i < pxd_num_contexts; ++i) {
+		ctx = &pxd_contexts[i];
+		if (ctx->num_devices == 0) {
+			continue;
+		}
+		printk(KERN_INFO "%s: pxd_ctx: %s ndevices: %lu",
+			__func__, ctx->name, ctx->num_devices);
+		printk(KERN_INFO "\tFC: connected: %d", ctx->fc.connected);
+	}
+	return 0;
+}
+
+static long pxd_ioctl_get_version(void __user *argp)
+{
+	char ver_data[64];
+	int ver_len = 0;
+
+	if (argp) {
+		ver_len = strlen(gitversion) < 64 ? strlen(gitversion) : 64;
```

Comment:
I see that the version string now includes branch info as well, if that is causing this overflow, do you want to trim the branch and fetch only the git commit. with the current length based truncate, it may not hold the full commit id.

Reply:
I did not change any logic, this change only moves the code around.